gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsBatch.java

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.dns;
+
+import static java.net.HttpURLConnection.HTTP_NOT_FOUND;
+
+import com.google.api.client.googleapis.json.GoogleJsonError;
+import com.google.api.services.dns.model.Change;
+import com.google.api.services.dns.model.ChangesListResponse;
+import com.google.api.services.dns.model.ManagedZone;
+import com.google.api.services.dns.model.ManagedZonesListResponse;
+import com.google.api.services.dns.model.Project;
+import com.google.api.services.dns.model.ResourceRecordSet;
+import com.google.api.services.dns.model.ResourceRecordSetsListResponse;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.gcloud.Page;
+import com.google.gcloud.PageImpl;
+import com.google.gcloud.dns.spi.DnsRpc;
+import com.google.gcloud.dns.spi.RpcBatch;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A batch of operations to be submitted to Google Cloud DNS using a single RPC request.
+ */
+public class DnsBatch {
+
+  private RpcBatch batch;
+  private final DnsRpc dnsRpc;
+  private final DnsOptions options;
+
+  DnsBatch(DnsOptions options) {
+    this.options = options;
+    this.dnsRpc = options.rpc();
+    this.batch = dnsRpc.createBatch();
+  }
+
+  Object batch() {
+    return batch;
+  }
+
+  DnsRpc dnsRpc() {
+    return dnsRpc;
+  }
+
+  DnsOptions options() {
+    return options;
+  }
+
+  /**
+   * Adds a request representing the "list zones" operation to this batch. The {@code options} can
+   * be used to restrict the fields returned or provide page size limits in the same way as for
+   * {@link Dns#listZones(Dns.ZoneListOption...)}. The returned {@link DnsBatchResult} will return a
+   * page of zones upon calling {@link DnsBatchResult#get()} on successful completion, or it will
```

Comment:
Up to you, but you can make the last sentence more concise by saying 

"Calling {@link DnsBatchResult#get()} on the return value yields a page of zones if successful and throws a {@link DnsException} otherwise."

If you decide it's too much effort to change here and in the methods below, I totally get it, it's a nit phrasing comment.